### PR TITLE
Allowing an array of `resources` config values

### PR DESCRIPTION
**Pull request by stephencookdev**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -147,7 +147,29 @@ Moves fonts from src to dest
...
 ### resources
 
-Moves miscellaneous resources from src to dest
+Moves miscellaneous resources from src to dest. It can accept an array of src/dest objects
+
+#### Options
+
+ * `src`
+ * `dest`
+ * `fileGlob` - the glob to search for files under the `src` directory (`**` by default)
```
Comment (rfreitas) [inline README.md]: any reason for picking `fileGlob`? I feel something like `filePattern` might fit it better, what do you think?

File: tasks/resources.js [hunk 2]
```diff
@@ -4,23 +4,37 @@ var changed = require('gulp-changed');
...
+    };
+  });
 
   var resourcesTask = function() {
-    gutil.log('Building resources from ' + JSON.stringify(paths.src));
+    gutil.log('Building resources from ' + JSON.stringify(pathsList.map(function(paths) {
+      return paths.src;
+    })));
+
+    var streams = pathsList.map(function(paths) {
```
Comment (rfreitas) [inline tasks/resources.js]: MINOR: a `pathsList` with `paths` inside feels a bit weird, but I can't think of a better name anyway.

File: tasks/watch.js [hunk 3]
```diff
@@ -15,11 +15,18 @@ module.exports = function(config) {
...
 
     watchableTasks.forEach(function(taskName) {
-      var task = config.tasks[taskName];
-      if(task) {
-        var taskSrc = Array.isArray(task.src) ? '{' + task.src.join(',') + '}' : task.src;
-        var glob = path.join(config.root.src, taskSrc, '**/*');
-        watch(glob, { usePolling: true }, function() {
+      var taskList = config.tasks[taskName];
+      if(taskList) {
+        if(!Array.isArray(taskList)) {
```
Comment (rfreitas) [inline tasks/watch.js]: I have seen this pattern so much, that I wonder whether it shouldn't be abstracted into a function.

File: tasks/resources.js [hunk 4]
```diff
@@ -4,23 +4,37 @@ var changed = require('gulp-changed');
...
-    src: path.join(config.root.src, config.tasks.resources.src, '/**'),
-    dest: path.join(config.root.dest, config.tasks.resources.dest)
-  };
+  if(!Array.isArray(resources)) {
+    resources = [resources];
+  }
+
+  var pathsList = resources.map(function(res) {
+    return {
+      src: path.join(config.root.src, res.src, res.glob || '**'),
```
Comment (rfreitas) [inline tasks/resources.js]: shouldn't it be called `fileGlob` instead of `glob`?

File: tasks/watch.js [hunk 5]
```diff
@@ -15,11 +15,18 @@ module.exports = function(config) {
...
+          taskList = [taskList];
+        }
+
+        var globList = taskList.map(function(task) {
+          var taskSrc = Array.isArray(task.src) ? '{' + task.src.join(',') + '}' : task.src;
+          return path.join(config.root.src, taskSrc, '**/*');
```
Comment (rfreitas) [inline tasks/watch.js]: couldn't/shouldn't we use the `fileGlob` here too?

File: README.md (hunk 1 above)
Comment (stephencookdev) [inline README.md]: When I hear 'pattern' I think more like regex etc.

But it's specifically a glob pattern

File: tasks/resources.js (hunk 2 above)
Comment (stephencookdev) [inline tasks/resources.js]: I agree with you, but also can't think of anything particularly great :(

File: tasks/watch.js (hunk 5 above)
Comment (stephencookdev) [inline tasks/watch.js]: We should do really - but it was never using them here before, and I was being lazy.

I'll add it in

File: tasks/watch.js (hunk 3 above)
Comment (stephencookdev) [inline tasks/watch.js]: Do you mean in general, or in pinion specifically?

After the recent diff change, I've just grepped through the code-base and there's only one example of this pattern remaining in pinion

File: tasks/resources.js (hunk 2 above)
Comment (stephencookdev) [inline tasks/resources.js]: This block of code has been refactored out now :)

File: tasks/watch.js (hunk 5 above)
Comment (stephencookdev) [inline tasks/watch.js]: The watch task now checks directory structures using the same config as the build tasks

File: tasks/resources.js (hunk 4 above)
Comment (stephencookdev) [inline tasks/resources.js]: It should have been, yes. This has been corrected in the refactor of this PR

File: README.md (hunk 1 above)
Comment (stephencookdev) [inline README.md]: I can't really find solid examples either way (most places just depend entirely on `src`, which is maybe something we should do to simplify the config values, but would be a breaking change, so not for this PR)

But the actual library that gulp comes down on to evaluate these patterns is [node-glob](https://github.com/isaacs/node-glob), which is much more pro "glob" than "pattern" in terms of terminology

**Pull request merged**